### ISSUE TITLE
Calculating METEOR CIDEr BLEU scores

### ISSUE THREAD

**Issue by arjung128** [lukemelas/image-paragraph-captioning#3]
Once the model has been trained, how can the METEOR CIDEr BLEU-1 BLEU-2 BLEU-3 BLEU-4 results presented in the paper be obtained? 

Also, does this implementation in its default state use repetition penalty? How can this setting be changed?

**Comment by lukemelas** [comment]
The scores are calculated every validation step (on the validation set). 

You can use the `block_trigrams` and `alpha` parameters to set the repetition penalty: 
```
python train.py  ...  other parameters  ...  --block_trigrams 1  --alpha 10.0
```

**Comment by arjung128** [comment]
How do you run the trained model on the validation set?

**Comment by lukemelas** [comment]
When you train the model, it will run on the validation set after every epoch.

**Comment by arjung128** [comment]
When I trained the model, I was getting just the CIDEr scores (many per epoch), and even those were fluctuating quite a bit (as in the scores were all over the place for all epochs, there wasn't an obvious (increasing) trend). I was not getting the METEOR or BLEU scores. Is there a way to get a single value for all six (METEOR CIDEr BLEU 1 - 4) metrics for the final model?

**Comment by lukemelas** [comment]
I am not sure, but I believe you're talking about the CIDEr scores for each batch. Those are printed for every batch and they are expected to vary widely. The final CIDEr/METEOR/BLEU scores on the validation set are printed once per batch, at the end of the epoch. Those should have clear increasing trend (if you wait a couple of epochs -- sometimes self-critical training is weird in that the metrics sometimes decrease a bit before going up).

**Comment by arjung128** [comment]
For some reason, the final CIDEr/METEOR/BLEU scores on the validation set are not printed at the end of each epoch when I train. Here's what my output looks like when I train:

...
Cider scores: 0.05055747138711029
Cider scores: 0.23762433183792392
Cider scores: 0.2016461526656507
Cider scores: 0.09484974868502609
Cider scores: 0.06714491373311673
Read data: 1.02520489693
iter 42201 (epoch 28), avg_reward = -0.011, data_time = 0.014, time/batch = 1.025
Cider scores: 0.12018177868980709
Cider scores: 0.1537472398105469
Cider scores: 0.02683920305320616
Cider scores: 0.17457595857620622
Cider scores: 0.13120042844450647
...
Cider scores: 0.14903162980352952
Cider scores: 0.1480045371928352
Cider scores: 0.1597950489871271
Cider scores: 0.0681437261069909
Cider scores: 0.05616059625852486
Read data: 1.14169120789
iter 42401 (epoch 29), avg_reward = 0.016, data_time = 0.014, time/batch = 1.142
Cider scores: 0.13915968413271335
Cider scores: 0.2236347322797569
Cider scores: 0.11668309405737172
Cider scores: 0.3331930390082059
Cider scores: 0.0617720056876462
...

Just a bunch of Cider scores per batch, but no CIDEr/METEOR/BLEU between epochs...Is there any way to change this so that the the final CIDEr/METEOR/BLEU scores on the validation set are printed once per batch, at the end of the epoch? Or is there a way to calculate the final CIDEr/METEOR/BLEU scores on the validation set given the final checkpoint generated by training?
